demux/packet.c

```diff
@@ -209,6 +211,30 @@ size_t demux_packet_estimate_total_size(struct demux_packet *dp)
     if (dp->avpacket) {
         mp_assert(!dp->is_cached);
         size += ROUND_ALLOC(dp->len);
+        if (dp->is_wrapped_avframe) {
+            mp_require(dp->buffer);
+
+            const AVFrame *frame = (AVFrame *)dp->buffer;
+            if (frame->hw_frames_ctx) {
+                const AVHWFramesContext *hwctx = (AVHWFramesContext *)frame->hw_frames_ctx->data;
+                size += ROUND_ALLOC(av_image_get_buffer_size(hwctx->sw_format, hwctx->width,
+                                                             hwctx->height, 16));
+            }
+            for (int i = 0; i < MP_ARRAY_SIZE(frame->buf) && frame->buf[i]; ++i)
+                size += ROUND_ALLOC(frame->buf[i]->size);
+
+            size += ROUND_ALLOC(frame->nb_extended_buf * sizeof(frame->extended_buf[0]));
```

Comment:
`nb_extended_buf` is audio-only, and this will already fail for that.

Reply:
Actually, how do you plan on handling this if the stream isn't AVMEDIA_TYPE_VIDEO? If you don't, then `extended_buf` isn't used for video frames, per the documentation in avutil/frame.h

Reply:
What's the problem? If it is unused the `nb_extended_buf` will be 0.

Reply:
Extra code that will never execute isn't a good thing. If audio is permitted and this isn't superfluous, then you'll also need to check the size of the channel layout field, which is another bit of code to handle.